presentation/src/main/java/com/anwera64/pagodividido/newexpenditure/NewExpenditureActivity.kt

```diff
@@ -0,0 +1,187 @@
+
+import android.app.AlertDialog
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.ArrayAdapter
+import android.widget.LinearLayout
+import androidx.annotation.StringRes
+import androidx.core.widget.doOnTextChanged
+import androidx.lifecycle.Observer
+import com.anwera64.pagodividido.R
+import com.anwera64.pagodividido.base.BaseViewModelActivity
+import com.anwera64.pagodividido.databinding.ActivityNewExpenditureBinding
+import com.anwera64.pagodividido.trip.TripActivity
+import com.anwera64.pagodividido.utils.EventWrapper
+import com.anwera64.pagodividido.utils.ViewUtils
+import com.anwera64.pagodividido.utils.nullOrHandled
+import com.anwera64.pagodividido.utils.uicomponents.DebtorInputView
+import com.anwera97.domain.models.CompanionModel
+import com.anwera97.domain.models.DebtorInputError
+import com.anwera97.domain.models.DebtorInputErrorReasons
+import com.anwera97.domain.models.InputErrorTypes
+import com.google.android.material.textfield.TextInputLayout
+
+class NewExpenditureActivity :
+    BaseViewModelActivity<NewExpenditureViewModel, ActivityNewExpenditureBinding>(
+        NewExpenditureViewModel::class
+    ) {
+
+    companion object {
+        private const val NOT_FOUND = -1
+    }
+
+    override val viewModelValue: Int? = null
+    override val layout: Int = R.layout.activity_new_expenditure
+    private val debtorInputs = ArrayList<DebtorInputView>()
+    private var payerId = NOT_FOUND
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setupUi()
+        viewModel.getCompanions(getTripId())
+    }
+
+    override fun setupObservers() {
+        viewModel.companions.observe(this, Observer(::onCompanionsObtained))
+        viewModel.debtInputErrors.observe(this, Observer(::observeDebtorInputErrors))
+        viewModel.showMatchErrorDialog.observe(this, Observer(::observeMismatchAlertDialog))
+        viewModel.amountInputError.observe(this, Observer(::setAmountInputError))
+        viewModel.payerInputError.observe(this, Observer(::setPayerInputError))
+        viewModel.insertonDone.observe(this, Observer(::onInsertionDone))
+    }
+
+    private fun onInsertionDone(eventWrapper: EventWrapper<Boolean>) {
+        if (eventWrapper.nullOrHandled()) return
+        if (eventWrapper.getContentIfHandled() == true) finish()
+    }
+
+    private fun getTripId() = intent.getIntExtra(TripActivity.TRIP_ID, -1)
+
+    private fun setupUi() {
+        setSupportActionBar(binding.toolbar)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    private fun loadSpinnerData(nameList: List<CompanionModel>) = with(binding.tiPayer) {
+        val arrayAdapter = ArrayAdapter(context, R.layout.list_metrial_drop_down_item, nameList)
+        setAdapter(arrayAdapter)
+        doOnTextChanged { text, _, _, _ ->
+            nameList.find { companion -> companion.name == text.toString() }
```

Comment:
we can rename the companion name to make it more readable

Reply:
What do you mean?